spec/factories/orchestration_template.rb

```diff
@@ -1,5 +1,17 @@
 FactoryGirl.define do
-  factory :orchestration_template do
+  # A solid class to implement require methods
+  # Cannot stub these methods because the class is used to test copying
+  class OrchestrationTemplateTest < OrchestrationTemplate
+    def validate_format
+      nil
+    end
+
+    def parameter_groups
+      []
+    end
+  end
+
+  factory :orchestration_template, :class => "OrchestrationTemplateTest" do
```

Comment:
@chrisarcand @jrafanie The base class `OrchestrationTemplate` has these method implemented as `raise NotImplementedError` and are expected to be overridden in the subclass.  Is this a reasonable solution for testing or can you recommend another approach?

Reply:
I don't know.  I can't really understand what's happening here.  Hardcoded test class names are sometimes a source of sporadic test failures .  With that said, I don't understand why this is needed.  I don't know what the comment on line 3 means either.

Where are we testing the base (abstract) class and the concrete subclasses?
Why can't we stub the methods?  What does it mean to test copying?